Updated wiki page (#5)
strongbox/strongbox-docs: docs/developer-guide/project-structure.md

# Project structure
## Modules
### Core Modules

All the core modules are located under the [Strongbox] project. Each of these modules has a `README.md` file explaining briefly what kind of code contains. New modules should also follow the same format.

This is a brief breakdown of the modules:

> * [strongbox-aql](https://github.com/strongbox/strongbox/tree/master/strongbox-aql)

Reviewer: Could please change the links in this page to use link references? It's easier to read and modify in the future.

```suggestion
* [strongbox-aql]
// and then somewhere in the footer
[strongbox-aql]: https://github.com/strongbox/strongbox/tree/master/strongbox-aql
```

>     This is where the [Artifact Query Language (AQL)](https://strongbox.github.io/user-guide/artifact-query-language.html) is located.

Reviewer: ```suggestion
    This is where the code for [Artifact Query Language (AQL)] is located.
```

    
* [strongbox-client]    
    This is where the artifact client resides.

* [strongbox-commons] 
    This is where the most common code that is shared across modules resides.

> * [strongbox-configuration]

Reviewer: There is a missing link here. Also, would you mind making the text for all of the modules in this document on a new line as they were before? This is done by adding a 2 white spaces at the end:
 ```suggestion
* [strongbox-configuration]  
```

    Contains configuration parsing related code.

> * [strongbox-cron]

Reviewer: There is a missing link here

>     * [strongbox-cron-api]

Reviewer: There is a missing link here

        Contains the Cron API code, that is required by custom cron and controller implementations.

>     * [strongbox-cron-tasks]

Reviewer: There is a missing link here

        Contains the common built-in cron tasks.

> * [strongbox-data-service]

Reviewer: There is a missing link here

    Contains the base implementation of data service classes. Check the [Getting Started With Persistence](https://strongbox.github.io/developer-guide/getting-started-with-persistence.html) article for more details.

> * [strongbox-distribution]

Reviewer: There is a missing link here. Also, I think we can safely move the text from here:

```suggestion
* [strongbox-distribution]  
    This module produces the final binary distributions which are then made public under the [releases](https://github.com/strongbox/strongbox/releases) section.
```

Note: I'm using a link here, because `releases` is a common word and if you set a reference link this will make all `releases` words in the text as links.

* [strongbox-event-api]
    Contains our [Event API](https://strongbox.github.io/developer-guide/using-the-event-api.html). 

* [strongbox-resources]

    * [strongbox-common-resources]
        This is the place where common resources which can be used by multiple projects reside. The idea is not have to duplicate things such as `logback*xml`, keystores and so on across the other modules. These resources are copied using the `maven-dependency-plugin`.

    * [strongbox-storage-api-resources]
        Contains the `strongbox.xml` configuration file.

* [strongbox-rest-client]
    Contains the REST API client.

* [strongbox-security]

    * [strongbox-authentication-api]

    * [strongbox-authentication-providers]

        * [strongbox-default-authentication-provider]

        * [strongbox-ldap-authentication-provider]

    * [strongbox-authentication-registry]

    * [strongbox-authentication-support]

    * [strongbox-security-api]
        Contains various security and encryption related classes.

    * [strongbox-user-management]

* [strongbox-storage]
    Contains the code for the storage related modules.

    * [strongbox-storage-api]
        Contains the the storage API.

>     * [strongbox-storage-core]

Reviewer: There is a missing link here

        Contains the core classes for the storage API.
        

>     * [strongbox-storage-layout-providers]

Reviewer: There is a missing link here

>         * [strongbox-storage-maven-layout]

Reviewer: There is a missing link here

>             * [strongbox-maven-metadata-api]

Reviewer: There is a missing link here

                This is the implementation of the support for the `maven-metadata.xml` format.

>             * [strongbox-storage-maven-layout-provider]

Reviewer: There is a missing link here

>                 This is the implementation of the [Maven 2 layout provider][Maven Layout Provider](https://strongbox.github.io/developer-guide/layout-providers/nuget-layout-provider.html). It depends on the [strongbox-maven-metadata-api](https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api).

Reviewer: ```suggestion
                This is the implementation of the [Maven Layout Provider]. It depends on the [strongbox-maven-metadata-api].
```

Typo: The `[Maven Layout Provider]` link should point to the `maven-layout-provider` page, not the `nuget` one :)

>     * [strongbox-storage-npm-layout-provider]

Reviewer: There is a missing link here

        This is the implementation of the [NPM layout provider](https://strongbox.github.io/developer-guide/layout-providers/npm-layout-provider.html). It depends on the [strongbox-npm-metadata](https://github.com/strongbox/strongbox-npm-metadata) project.

>     * [strongbox-storage-nuget-layout-provider]

Reviewer: There is a missing link here

>         This is the implementation of the [Nuget layout provider](https://strongbox.github.io/developer-guide/layout-providers/npm-layout-provider.html).

Reviewer: The Nuget Layout Provider link is pointing to the wrong page.
```suggestion
        This is the implementation of the [Nuget layout provider]
        // bottom of the page
        [Nuget layout provider]: https://strongbox.github.io/developer-guide/layout-providers/nuget-layout-provider.html
```

>     * [strongbox-storage-p2-layout-provider]

Reviewer: There is a missing link here

        This is an incomplete early draft implementation of the P2 OSGi layout provider. This module needs a lot more work, before the P2 layout provider could be usable.

>     * [strongbox-storage-raw-layout-provider]

Reviewer: There is a missing link here

        This is the implementation of the [Raw layout provider](https://strongbox.github.io/developer-guide/layout-providers/raw-layout-provider.html).
        
* [strongbox-testing]
    Contains various very useful base classes for testing.

    * [strongbox-testing-core]
        Contains common testing-related code.

>     * [strongbox-testing-storage]

Reviewer: There is a missing link here

    * [strongbox-testing-web]
        Contains the common `restassured` configuration for controller tests.

* [strongbox-web-core]
    This is the web module which contains all the controllers.

> * [strongbox-web-forms]

Reviewer: There is a missing link here

>     This is the module that contains all the web forms. You might find the [Writing Web Form Tests](https://strongbox.github.io/developer-guide/writing-web-form-tests.html) useful.

Reviewer: This should not point to the domain. It should use a relative path to the `writing-web-form-tests.md` file. Then `mkdocs` will automatically create the right link.

```suggestion
    This is the module that contains all the web forms. You might find the [Writing Web Form Tests] page useful.
    // bottom of the page
    [Writing Web Form Tests]: ./writing-web-form-tests.md
```

[strongbox-parent]: https://github.com/strongbox/strongbox-parent/tree/master
[strongbox-resources]: https://github.com/strongbox/strongbox/tree/master/strongbox-resources
[strongbox-rest-client]: https://github.com/strongbox/strongbox/tree/master/strongbox-rest-client
[strongbox-security]: https://github.com/strongbox/strongbox/tree/master/strongbox-security

> [strongbox-security-api]: https://github.com/strongbox/strongbox/tree/master/strongbox-security-api

Reviewer: Wrong link:
```suggestion
[strongbox-security-api]: https://github.com/strongbox/strongbox/tree/master/strongbox-security/strongbox-security-api
```

[strongbox-storage]: https://github.com/strongbox/strongbox/tree/master/strongbox-storage
[strongbox-storage-api]: https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-api

> [strongbox-storage-api-resources]: https://github.com/strongbox/strongbox/tree/master/strongbox-resources/strongbox-storage-resources/strongbox-storage-api-resources

Reviewer: The link is wrong:
```suggestion
[strongbox-storage-api-resources]: https://github.com/strongbox/strongbox/tree/master/strongbox-resources/strongbox-storage-api-resources
```
